### ISSUE TITLE
color eq slider names

### ISSUE THREAD

**Issue by SoupyGit** [darktable-org/darktable#16260]
The name of some sliders in color eq will naturally be subjective since they cover multiple hues, however other sliders cover a much narrower range, and that range is not reflected by their current name.

To demonstrate, I am working with this hue circle:

![all](https://github.com/darktable-org/darktable/assets/69561443/fb0c4c6a-4466-4a00-b55f-ce20f4eaa692)

1) This one is called 'turqoise', yet you will see in both the hue circle and scope that it perfectly aligns with cyan, and does not even include turqoise. It should be called 'cyan':

 ![cyan](https://github.com/darktable-org/darktable/assets/69561443/ae26e1cb-9ac6-4d6d-9a10-36a58afc024d)

The actual colour turqoise, which is in between cyan and green, is clearly a part of the 'green' range:

![green](https://github.com/darktable-org/darktable/assets/69561443/22194eb5-ff4f-4f8b-958c-49782d996448)

2) This one is called 'blue', yet does not act on the primary colour blue, but the range in between blue and cyan. The correct name is 'teal':

![teal](https://github.com/darktable-org/darktable/assets/69561443/c3aca54a-f495-4a98-ac3d-86058f8e0bef)

3) This one is called 'lavender', which is ok since it does include that colour (could also be called 'purple' or 'violet'), however is one of the subjective ones, since its range also covers the primary colour blue. Since blue is primary and lavender is tertiary, and thus is more likely to be sort after by editors, I would propose the name be changed to 'blue':

![blue](https://github.com/darktable-org/darktable/assets/69561443/aee1639d-2a3f-4fef-afc9-d76fc0b612e0)

4) This one is called 'purple', yet as seen in both hue circle and scope, aligns perfectly with 'magenta'. It should be called 'magenta':

![magenta](https://github.com/darktable-org/darktable/assets/69561443/3f8c321d-b655-49b6-a0cd-4b240a1a13cc)

5) This one is called 'lime', and is another subjective one as its range also covers 'yellow'. I don't have a firm opinion:

![lime-yellow](https://github.com/darktable-org/darktable/assets/69561443/53646755-1042-421c-88a1-6e8c79cf0174)

**Comment by jenshannoschwalm** [comment]
In general i agree on "some names are not good". Just took those from original author.

1. purple -> magenta? **yes**
2. turqoise -> cyan? **yes**
3. all blue changes - not sure

**Comment by Christian-Bouhon** [comment]
Hello,

Very nice analysis, you beat me to it ;-).
I made a few test patterns to better visualize the colors and the action of the various settings.
I found that the problem lies more in the distribution of colors. I find the angle of the colors turquoise and blue too narrow and the angle of the color lavander too wide.

https://github.com/darktable-org/darktable/assets/120166629/f0200166-fa20-45d8-b438-b0f449fc12c6

You'll find the various test images here [https://www.dropbox.com/scl/fi/awwnbhpg4ux2jnglncbyf/Mire-HSL.zip?rlkey=orsshzbhmi1756ntngckfqueu&dl=0](url)

@jenshannoschwalm @TurboGit You are doing a magnificent job of devolleppment, thank you.

Greetings from Brussels.
Christian

**Comment by SoupyGit** [comment]
@jenshannoschwalm Yes those are the most obvious two.

The name teal is certainly debatable, as many different sites seem to have a different opinions on what teal is. My reasoning was that rgb 255 128 0 is orange. Teal and orange is a famous analogous harmony, and the opposite of orange is rgb 0 128 255, thus should be 'teal'. However, according to wikipedia the name teal comes from a bird with that colour on its head, which does look greenish, closer to turqoise, and therefore 'teal' wouldn't be a suitable name for the range of this slider.

The name most commonly found for that rgb trio is 'azure', so that one seems better.

If you don't change 'lavender' it might not be necessary to change 'blue' to 'teal/azure', however it doesnt seem correct to name a slider after a primary colour it doesnt contain.

'Green' slider range also contains turqoise and we name it after the primary, so it would be consistent to do the same for 'blue/lavender' and call it 'blue'.

Just my 2c.

**Comment by todd-prior** [comment]
Names could align with those often used to illustrated the names used with mixing as noted in this document about the channel mixer...
[The Channel Mixer.pdf](https://github.com/darktable-org/darktable/files/14188306/The.Channel.Mixer.pdf)

**Comment by todd-prior** [comment]
> 3\. all blue changes - not sure

Why does the blue slider not act on a range of hues around the blue primary as red and green do around their respective locations ... its very clear in the vector scope that it is quite different... I think at first I thought it was maybe a color space things but I changed a few and that didn't seem to make a huge difference

**Comment by SoupyGit** [comment]
@todd-prior that's a document I put together. As above, I'm thinking now azure (instead of teal) is a more accurate and established name for that rgb colour.

**Comment by todd-prior** [comment]
Prior,Todd reacted to your message:

**Comment by jenshannoschwalm** [comment]
Revision of above comment:
1. purple -> magenta? **yes**
2. turqoise -> cyan? **yes**
3. blue ->azure ? **yes**

**Comment by SoupyGit** [comment]
An alternate idea is to have unequal spacing between points. That would allow to merge 'azure' and 'cyan' (which is currently very narrow) into one, and split 'blue' from 'lavender' into two. It would still be eight points, just spaced differently.

**Comment by jenshannoschwalm** [comment]
> An alternate idea is to have unequal spacing between points. That would allow to merge 'azure' and 'cyan' (which is currently very narrow) into one, and split 'blue' from 'lavender' into two. It would still be eight points, just spaced differently.

Nope for sure. Has been discussed in the original PR and AP's arguments are absolutely valid on this.

**Comment by jenshannoschwalm** [comment]
@TurboGit @s7habo @elstoc, any opinion on this?

**Comment by s7habo** [comment]
I don't have a strong opinion on this either. 

You can see which colors are influenced in the background. I think this visualization is crucial. Besides, this is not a module with which you should precisely cover something like pantone color chart but is rather intended for rudimentary changes. 

So for me, everything you agree on fits.

**Comment by todd-prior** [comment]
> Nope for sure. Has been discussed in the original PR and AP's arguments are absolutely valid on this. And, a) nodes are all 45°-spaced and b) guided filter depends on this.

Would you have the number or link of the orginal PR... I'll try to find it but in case I don't...no rush and still I feel like unless there is some color space mapping that puts "blue" where blue is that it act barely on blue and so given that red and green seem fairly aligned how can blue come to land where it does .... maybe the PR explains it and if so don't bother wasting your time on me....

**Comment by elstoc** [comment]
Honestly I don't care what we call the colours, and I'd much prefer not to display those sliders at all. I now know I can hide them but the fact we've chosen to have that (hide) option suggests to me that we suspect they might be pointless. Same goes for the sliders in the tone equalizer. Do they add anything? No

Another advantage of removing the names is that we could do away with the fixed nodes. I understand we need to keep the nodes equidistant but if we could shift them all left/right at the same time to place a colour of interest directly under a node that would be great.

**Comment by jenshannoschwalm** [comment]
@elstoc i agree with you on removing the sliders.

**Comment by ptilopteri** [comment]
* Hanno Schwalm ***@***.***> [02-08-24 16:21]:
> @elstoc i agree with you on removing the sliders.
> 
> -- 
> Message ID: ***@***.***>

me too
-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
http://en.opensuse.org    openSUSE Community Member    facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by Christian-Bouhon** [comment]
Personally, I like the current interface, and the fact that you can choose to show or hide the sliders. I find it more user-friendly to work with the sliders, which allows me to make finer adjustments (same goes for the tone equalizer). I admit it's all very subjective.

As for the color names, I'd keep the blue name even though the node isn't positioned on the primary color (240°).

**Comment by SoupyGit** [comment]
I don't have a strong opinion about keeping or removing sliders, but even if they are removed, the name of the hue point still shows up in the pop up label when you hover over, and will presumably be in the manual too, so is still relevant.

Although I think cyan and magenta are the two most obvious names to change, my main motivation to look into this was to see if some of the confusion people had around why the blue slider didn't affect blue could be resolved by more accurate naming. If the names of blue and lavender remain unchanged, that issue still remains. Removing the sliders (and also colour names from pop ups and manual) might be another solution. Since the graph shows colours many people wouldn't have a problem with that, though it wouldn't be very helpful for the colour blind.

> I understand we need to keep the nodes equidistant but if we could shift them all left/right at the same time to place a colour of interest directly under a node that would be great.

If that were to happen I guess they would be named numerically, 1 to 8.

**Comment by todd-prior** [comment]
When I experimented with a color checker moving the nodes seemed to create a decent match between the changed color in the image is the patch and what was indicated as the change in the eq graph eg yellow would move to resemble the color on
 the EQ graph... except for blue.moving it left the node on a color not close to what was reflected 


Sent from 
Outlook for Android

**Comment by Christian-Bouhon** [comment]
Hello,
Yesterday, I used the module for a few photos with good, easy-to-get results :-).
As for the names, here's my proposal, taking into consideration the various proposals quoted above and the French translation.

1 red (Fr = Rouge): unchanged
2 Orange  : unchanged
3 Lime (Fr = **Vert** citron) : Lemon (Fr = Citron)
4 Green (Fr = **Vert**) : unchanged
5 Turqoise (Fr = Turquoise): unchanged or Cyan
6 Blue (Fr = Bleu): unchanged
7 Lavander (Fr = Lavande) : unchanged
8 Purple (fr = Violet) : unchanged or Magenta

In conclusion, in the end I'd only change lime to lemon, what makes me hesitate for cyan and magenta is that they're well-defined colors (0,255,255) and (255,0,255). I'd give a preference to more subjective color names.
Best regards,
Christian

**Comment by MStraeten** [comment]
> @elstoc i agree with you on removing the sliders.

removing the sliders needs to make the nodes in the graph accessible for shortcuts. So the current show/hide solution is quite more useful

**Comment by jenshannoschwalm** [comment]
We could just hide them from presentation....

**Comment by jenshannoschwalm** [comment]
I think there is no reason to discuss further. We'll keep them, enough people want them and it doesn't hurt ...

**Comment by jenshannoschwalm** [comment]
Closing this for now. UI has some significant progress so "naming" should be discussed in another FR if new arguments are coming ...
